### DISCUSSION TITLE
Verstappen data from second day of PS Testing...

### DISCUSSION THREAD

**Discussion by ricky732** [theOehrly/Fast-F1#109] [General]
Getting this error if I try to visualyze Verstappen data (driver: 'VER')... Is that a bug of Max's datas or is that something wrong by me? With every other driver, the code works fine...

[
![issue fastf1](https://user-images.githubusercontent.com/46622560/157928102-537e800b-77b2-4e3e-bb2f-9382c41336dd.png)
](url)

**Comment by theOehrly** [comment]
The same code does work for other drivers? The error itself is not very conclusive. It seems that there are no laps for Verstappen. Did he drive on that day?

**Reply by ricky732** [reply]
I was looking for testing session (2022, 2, 2) so today... Today Max drove for RB if I am not stupid ahahaha.. Maybe you could try to plot his telemetry and see if you get the same error, maybe the live timing data for him are bugged and it doesn't read the laps he has done 😂

**Comment by bambz96** [comment]
I think this is because his driver number is not being resolved correctly, as he changed from 33 to 1 this year. It's working for me if I pick Verstappen by number 1 rather than the code 'VER'.

**Reply by theOehrly** [reply]
Ah yes, Max changing from 33 to 1 causes this. I hadn't thought of this. There's a similar issue with other drivers. The reason is:
FastF1 currently requires a hard coded list of drivers to map driver names and teams to the driver number (for testing events only). This list is not up to date in the current release.
Among other things, this will be resolved in the next pre-release version by pulling this data from the api as well.

As a workaround, use driver numbers for now.

**Comment by MaxZec** [comment]
Hello, I've updated the library with the latest version, but I'm not able aby more to pull out data... Not even for Baharain. Buti was used to type the command line like python F1Laps.py 2021 Spain --drivers HAM LEC --sessions Q R.. Don't know what to type to pull out data from the tests..any help pls?

**Reply by pgroms** [reply]
Try to install v 2.2 https://github.com/theOehrly/Fast-F1/discussions/105 and use following command 
preseason = ff1.get_testing_session(2022, 2, 3)

**Comment by MaxZec** [comment]
thnx mate,

if i try to install it I got this error

[image: image.png]

Il giorno lun 14 mar 2022 alle ore 09:59 pgroms ***@***.***>
ha scritto:

**Reply by theOehrly** [reply]
The image is missing

**Reply by MaxZec** [reply]
my bad sorry.
![photo_2022-03-16_15-51-35](https://user-images.githubusercontent.com/95567617/158618621-396b6140-1c6b-4721-845e-e664618a4c60.jpg)

**Reply by theOehrly** [reply]
It looks like you haven't downloaded the file to this directory?

**Comment by MaxZec** [comment]
Do I need to leave the whole zip file in here?? Coz I've just unpacked the file in this folder

**Reply by theOehrly** [reply]
Well, as the `DIR` command shows, the file is not in this folder? Even if it was, you need to add the file ending as well in the pip command. Not just the name.

Later today, you'll get the stable release, then you can update as usual without needing to download a file

**Reply by ricky732** [reply]
@theOehrly is Verstappen expected to work normally and correctly for this weekend if i call him 'VER'? Otherwise I'd have to re-write all the code for label exceptions when I call him with number 1, to correctly print VER on the title / legend of the plot... I hope it will work fine as all the other drivers...

**Reply by MaxZec** [reply]
![IMG_20220318_215715_456](https://user-images.githubusercontent.com/95567617/159082373-8e5a54ee-763d-4661-a6f7-716279261f4b.jpg)
![IMG_20220318_215708_676](https://user-images.githubusercontent.com/95567617/159082380-f0fe2626-de0e-4804-be82-6496bbc4fbb9.jpg)
I've successfully updated tje program to the last version.. But for some reason it's not downloading data like there is no speed or time...whete am I wrong?

**Reply by markste-in** [reply]
I mean you haven't posted any code but I assume the issue is similar to https://github.com/theOehrly/Fast-F1/discussions/118#discussioncomment-2388750 
Instead of `get_telemetry() `try `get_car_data()`

That worked for me
